Index: WebXml.java
===================================================================
RCS file: /usr/cvsroot/dpaplattform-mvn2/websearch/src/main/java/org/ajax4jsf/webapp/WebXml.java,v
retrieving revision 1.1
diff -u -p -F '\(public\|private\|protected\|#macro\)' -u -p -w -r1.1 WebXml.java
--- WebXml.java	20 Mar 2012 09:38:24 -0000	1.1
+++ WebXml.java	20 Mar 2012 09:51:27 -0000
@@ -22,6 +22,8 @@
 package org.ajax4jsf.webapp;
 
 import java.io.Serializable;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
 
 import javax.faces.context.FacesContext;
 import javax.servlet.ServletContext;
@@ -219,7 +221,16 @@ 	public String getFacesResourceKey(HttpS
 		String resourcePath = request.getRequestURI().substring(
 				request.getContextPath().length());// isPrefixMapping()?request.getPathInfo():request.getServletPath();
 
-		return getFacesResourceKey(resourcePath);
+		String s = getFacesResourceKey(resourcePath);
+		if (null != s) {
+			try {
+				return URLDecoder.decode(s, "ISO-8859-1");
+			} catch (UnsupportedEncodingException e) {
+				// Doesn't happen.
+			}
+		}
+
+		return null;
 	}
 
 	/**
